agents/dns/dns.py

```diff
@@ -18,3 +23,60 @@ def __init__(
             interface=interface,
             controller_url=controller_url,
         )
+
+        self.query_domain = query_domain
+
+    @staticmethod
+    def is_valid_ip(output):
+        """
+        Validate if the output is a correctly formatted IPv4 address.
+        Returns:
+            bool: True if the output is a valid IP address, False otherwise.
+        """
+        return re.match(r'^\d{1,3}(\.\d{1,3}){3}$', output.strip()) is not None
```

Comment:
Perhaps it's a good idea to have this utility function. I just feel like it would make more sense as a simple function somewhere in `agents/utils/helpers`. Moreover, maybe we'll even use it in other contexts.